### DISCUSSION TITLE
Wrong number of config param found!

### DISCUSSION THREAD

**Discussion by Whoajoe** [arkypita/LaserGRBL#1192] [Q&A]
I was trying to change a few settings to speed up my laser and they didn't save correctly.  Now my laser wont do anything.  How can I update the parameters if it wont read them?

LaserGrbl v3.8.0.0
Grbl v1.1f
Wrapper: UsbSerial
Win 10, Professional, 10.0.18363.0 (7)
CLR: C:\Windows\Microsoft.NET\Framework64\v2.0.50727\mscorlib.dll 2.0.50727.9151 (WinRelRS6.050727-9100)

TypeOf exception  [System.TimeoutException]
Exception message [Wrong number of config param found! (30/34)]
Exception source  [LaserGRBL], thread [Main Thread]
Exception method  [Void RefreshConfig()]

   ----------- stack trace -----------
   at LaserGRBL.GrblCore.RefreshConfig() in E:\LaserGRBL\LaserGRBL\GrblCore.cs:line 774
   at LaserGRBL.GrblConfig.BtnExport_Click(Object sender, EventArgs e) in E:\LaserGRBL\LaserGRBL\GrblConfig.cs:line 176
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by gmmanonymus111** [comment]
You could try sending `$RST=$`. That should reset $$-s to their default settings. (more info about reset [here](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Commands#rst-rst-and-rst--restore-grbl-settings-and-data-to-defaults))
If that won't reset your bouard, you need to re-flash GRBL if you messed up it's settings.
If it is only LaserGRBL, try to delete `C:\Users\{UserName}\AppData\Roaming\LaserGRBL\LaserGRBL.Settings.bin`

**Comment by arkypita** [comment]
LaserGRBL is exporting your grbl board configuration (because you press export button). The configuration is made by a few number of parameters stored inside the controller.

LaserGRBL try to read the config from the board, and know how many of this parameters to expect. It is expecting to receive 34 parameters, instead the board returns only 30 parameters. Don't know why.

Lasergrbl signal this issue.

What if you type $$ in "type gcode here"?
How many results you have?

**Comment by Whoajoe** [comment]
![image](https://user-images.githubusercontent.com/75652337/101525137-80712e00-3950-11eb-840d-6c2ca77c1004.jpeg)

Many things are zeroed out. I did the reset and it's back to factory settings. It's working correctly. I was trying to adjust the speed when the laser isn't on. I'm burning at 1,000 can the machine travel faster than that when laser is off?  It's a China 3018pro.

**Comment by arkypita** [comment]
LaserGRBL moves on white parts through the G0 command (unless you have changed the setting in raster import) which the machine interprets as a rapid movement performed at maximum speed and acceleration.

If you want your engraver to go faster during white shifts, simply change the configured maximum speeds and accelerations.

**Comment by ayalajw** [comment]
> ![image](https://user-images.githubusercontent.com/75652337/101525137-80712e00-3950-11eb-840d-6c2ca77c1004.jpeg)
> 
> Many things are zeroed out. I did the reset and it's back to factory settings. It's working correctly. I was trying to adjust the speed when the laser isn't on. I'm burning at 1,000 can the machine travel faster than that when laser is off? It's a China 3018pro.

how does one reset the lazer   email me please ayalajameswilson@gmail.com

**Comment by ayalajw** [comment]
please help
how can i reset thbe lazer {Ortur back to factory setings
pleases email me : ayalajameswilson@gmail.com

**Comment by arkypita** [comment]
send $RST=$ command via "type gcode here" box

**Comment by arkypita** [comment]
@Whoajoe I would be curious to know what causes all parameters to be reset. LaserGRBL cannot cause anything like that. did you use any other program?

**Comment by Whoajoe** [comment]
@ayalajw The $RST=$ worked for resetting my parameters. Once I reset back to factory I could then change the parameters to meet my needs. 

@arkypita The only software I was using was Lasergbrl. I don't think it was a software issue but more of a user issue. lol I think the speed I was trying to input was too fast and was out of parameters of what the machine could understand. I haven't tried it again but I think I was trying to get 6000 for my speed. I'm new to CNC stuff and I am still learning. I really enjoy it so far and I wish I can find an easy 3D software like this for the router.

**Comment by ayalajw** [comment]
Can you tell whast speed u up setting it at

Sent from Mail for Windows 10

**Comment by ayalajw** [comment]
How and where do I finally all this code stuff

**Comment by ayalajw** [comment]
Fine

**Comment by ayalajw** [comment]
Please text me
Ayalajeswilson@mail.com

**Comment by Whoajoe** [comment]
@ayalajw ![image](https://user-images.githubusercontent.com/75652337/101845676-e1496380-3b14-11eb-9d77-89943b511c32.jpeg)

For what it's worth this is what I have my set at. The laser seems to do a good job at this speed. It will make nice portraits burning at 1000 speed. It won't win no races tho.

**Comment by StuartB4** [comment]
@Whoajoe  Change $32 to 1

**Comment by ayalajw** [comment]
My neighbor help my wife when I was at home
  

  
  
  
    
  

  
  
>

**Comment by ayalajw** [comment]
Not good at this stuff
  
Where do I type that
  
  
  
    
  

  
  
>

**Comment by ayalajw** [comment]
Where do I type that
  
  
  
    
  

  
  
>

**Comment by StuartB4** [comment]
If you want to change the settings you have to make sure lasergrbl is connected. Then goto the Grbl menu and click Grbl configuration. Then you get the list of settings. If you change any of them you have to click the Write button to store the setting.

**Comment by Whoajoe** [comment]
@StuartB4 What does changing it to one do?

**Comment by StuartB4** [comment]
It enables laser mode . The easiest way to understand what it does is here:

"When laser mode is enabled, Grbl controls laser power by varying the 0-5V voltage from the spindle PWM D11 pin.
0V should be treated as disabled, while 5V is full power. Intermediate output voltages are also assumed to be linear with laser power, such that 2.5V is approximate 50% laser power.

M4 Dynamic Laser Power Mode:

Dynamic laser power mode will automatically adjust laser power based on the current speed relative to the programmed rate. It essentially ensures the amount of laser energy along a cut is consistent even though the machine may be stopped or actively accelerating. This is very useful for clean, precise engraving and cutting on simple materials across a large range of G-code generation methods by CAM programs. It will generally run faster and may be all you need to use.

Grbl calculates laser power based on the assumption that laser power is linear with speed and the material. Often, this is not the case. Lasers can cut differently at varying power levels and some materials may not cut well at a particular speed and/power. In short, this means that dynamic power mode may not work for all situations. Always do a test piece prior to using this with a new material or machine.

When not in motion, M4 dynamic mode turns off the laser. It only turns on when the machine moves. This generally makes the laser safer to operate, because, unlike M3, it will never burn a hole through your table, if you stop and forget to turn M3 off in time.

M3 Constant Laser Power Mode:
M3 can be used to keep the laser on for focusing.

Constant laser power mode simply keeps the laser power as programmed, regardless if the machine is moving, accelerating, or stopped. This provides better control of the laser state. With a good G-code program, this can lead to more consistent cuts in more difficult materials.
For a clean cut and prevent scorching with M3 constant power mode, it's a good idea to add lead-in and lead-out motions around the line you want to cut to give some space for the machine to accelerate and decelerate."

**Comment by ayalajw** [comment]
Thanks for that.   
  
  
  
  
  
    
  

  
  
>

**Comment by ayalajw** [comment]
EVENING
WHERE IN WINDOWS can I fine file to  reset my lazer GBL


Sent from Mail for Windows 10
